juspay/hyperswitch
# feat(payment_method): add capability to store bank details using /payment_methods endpoint

## Type of Change



## Description
Ref - Issue #3112 

### Additional Changes





## Motivation and Context



## How did you test it?

Tested locally

Postman collection - https://galactic-capsule-229427.postman.co/workspace/My-Workspace~2b563e0d-bad3-420f-8c0b-0fd5b278a4fe/collection/9906252-142ea0ea-f61f-48e7-8317-5f819dd535e9?action=share&creator=9906252

**Test Cases**
There are two major flows to be tested out which are fixed in this PR

1. Check payouts operation works end to end (payout creation + payout fulfillment for bank / card)
   - Create a payout with `auto_fulfill` false
   - Fulfill this payout using /payout/fulfill API call
   - This ensures the intermediary data is being persisted in temp locker
2. Payment methods saved for a customer can be used for any operation (payments / payouts) using payment_token
   - Create and fulfill a payout with `recurring` true against a cusomter_id
   - List SPMs for this customer
   - Using payment_token to carry out a payment or a payout transaction

## Checklist
